test/e2e/framework/workload.go

```diff
@@ -0,0 +1,91 @@
+package framework
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+
+	workloadv1alpha1 "github.com/karmada-io/karmada/examples/customresourceinterpreter/apis/workload/v1alpha1"
+)
+
+var workloadGVR = schema.GroupVersionResource{Group: "workload.example.io", Version: "v1alpha1", Resource: "workloads"}
+
+// CreateWorkload create Workload with dynamic client
+func CreateWorkload(client dynamic.Interface, workload *workloadv1alpha1.Workload) {
+	ginkgo.By(fmt.Sprintf("Creating workload(%s/%s)", workload.Namespace, workload.Name), func() {
+		unstructuredObj, err := runtime.DefaultUnstructuredConverter.ToUnstructured(workload)
+		gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+
+		_, err = client.Resource(workloadGVR).Namespace(workload.Namespace).Create(context.TODO(), &unstructured.Unstructured{Object: unstructuredObj}, metav1.CreateOptions{})
+		gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+	})
+}
+
+// UpdateWorkload update Workload's spec.paused with dynamic client
+func UpdateWorkload(client dynamic.Interface, namespace, name, clusterName string) {
+	ginkgo.By(fmt.Sprintf("Update workload(%s/%s) in cluster(%s)", namespace, name, clusterName), func() {
+		var err error
+		unstructuredObj := &unstructured.Unstructured{}
+
+		gomega.Eventually(func() error {
+			unstructuredObj, err = client.Resource(workloadGVR).Namespace(namespace).Get(context.TODO(), name, metav1.GetOptions{})
+			return err
+		}, pollTimeout, pollInterval).ShouldNot(gomega.HaveOccurred())
+
+		result, err := unstructuredObj.MarshalJSON()
+		gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+
+		workload := workloadv1alpha1.Workload{}
+		err = json.Unmarshal(result, &workload)
+		gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+
+		workload.Spec.Paused = true
+
+		newUnstructuredObj, err := runtime.DefaultUnstructuredConverter.ToUnstructured(&workload)
+		gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+
+		gomega.Eventually(func() error {
+			_, err = client.Resource(workloadGVR).Namespace(workload.Namespace).Update(context.TODO(), &unstructured.Unstructured{Object: newUnstructuredObj}, metav1.UpdateOptions{})
+			return err
+		}, pollTimeout, pollInterval).ShouldNot(gomega.HaveOccurred())
+	})
+}
+
+// GetWorkloadSpecPaused get WorkloadSpecPaused with dynamic client.
+func GetWorkloadSpecPaused(client dynamic.Interface, namespace, name string) bool {
```

Comment:
ditto, we can directly return the workload object to make this function more common.

Reply:
It is ok.